src/core/geometry/qgsgeometryutils.cpp

```diff
@@ -688,30 +749,58 @@ void QgsGeometryUtils::segmentizeArc( const QgsPoint &p1, const QgsPoint &p2, co
     if ( hasM )
       pointWkbType = QgsWkbTypes::addM( pointWkbType );
 
+    QgsDebugMsg( QString( "a1:%1 (%2), a3:%3 (%4), inc:%5, shi:?, cw:%6" )
+                 . arg( a1 ).  arg( a1 * 180 / M_PI )
+                 . arg( a3 ).  arg( a3 * 180 / M_PI )
+                 . arg( increment ).  arg( clockwise )
+               );
+
     //make sure the curve point p2 is part of the segmentized vertices. But only if p1 != p3
+    // TODO: make this a parameter
     bool addP2 = true;
     if ( qgsDoubleNear( circlePoint1.x(), circlePoint3.x() ) && qgsDoubleNear( circlePoint1.y(), circlePoint3.y() ) )
     {
       addP2 = false;
     }
-
-    for ( double angle = a1 + increment; angle < a3; angle += increment )
+    addP2 = false;
```

Comment:
@strk this has left a lot of unreachable code in the loop below - was this intended?